docs/reference/commandline/service_create.md

```diff
@@ -735,51 +735,75 @@ Containers on the same network can access each other using
 ### Publish service ports externally to the swarm (-p, --publish)
 
 You can publish service ports to make them available externally to the swarm
-using the `--publish` flag:
-
-```bash
-$ docker service create --publish <TARGET-PORT>:<SERVICE-PORT> nginx
-```
-
-For example:
+using the `--publish` flag. The `--publish` flag can take two different styles
+of arguments. The short version is positional, and allows you to specify the
+target port and container port separated by a colon.
 
 ```bash
 $ docker service create --name my_web --replicas 3 --publish 8080:80 nginx
 ```
 
-When you publish a service port, the swarm routing mesh makes the service
-accessible at the target port on every node regardless if there is a task for
-the service running on the node. For more information refer to
-[Use swarm mode routing mesh](https://docs.docker.com/engine/swarm/ingress/).
-
-### Publish a port for TCP only or UDP only
-
-By default, when you publish a port, it is a TCP port. You can
-specifically publish a UDP port instead of or in addition to a TCP port. When
-you publish both TCP and UDP ports, Docker 1.12.2 and earlier require you to
-add the suffix `/tcp` for TCP ports. Otherwise it is optional.
-
-#### TCP only
-
-The following two commands are equivalent.
+There is also a long format, which is easier to read and allows you to specify
+more options. The long format is preferred. You cannot specify the service's
+mode when using the short format. Here is an example of using the long format
+for the same service as above:
 
 ```bash
-$ docker service create --name dns-cache -p 53:53 dns-cache
-
-$ docker service create --name dns-cache -p 53:53/tcp dns-cache
+$ docker service create --name my_web --replicas 3 --publish target=8080,port=80 nginx
 ```
 
-#### TCP and UDP
+The options you can specify are:
 
-```bash
-$ docker service create --name dns-cache -p 53:53/tcp -p 53:53/udp dns-cache
-```
-
-#### UDP only
+<table>
```

Comment:
I personally prefer to create tables using [markdown syntax](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#tables).

Reply:
Me too! However, `docker/docker.github.io` uses these docs in a partial way (we grab the `## Description` and `## Examples` sections of the files) and our Markdown processor runs into trouble with Markdown tables in some  cases. It's a hell of a thing to debug, so in these reference we tend to favor HTML tables except for the most simple cases.